runtime/lua/vim/diagnostic.lua

```diff
@@ -1075,16 +1075,26 @@ local function goto_diagnostic(diagnostic, opts)
     vim.cmd('normal! zv')
   end)
 
-  local float_opts = opts.float
-  if float_opts then
+  if opts.float then
+    vim.deprecate('opts.float', 'opts.on_jump', '0.14')
+    local float_opts = opts.float ---@type table|boolean
     float_opts = type(float_opts) == 'table' and float_opts or {}
-    vim.schedule(function()
-      M.open_float(vim.tbl_extend('keep', float_opts, {
-        bufnr = api.nvim_win_get_buf(winid),
-        scope = 'cursor',
-        focus = false,
-      }))
-    end)
+
+    opts.on_jump = function(_, bufnr)
+      vim.schedule(function()
```

Comment:
Should we have the on_jump callback itself be executed in vim.schedule so that users don’t have to use vim.schedule in the callback?

I think it’s a reasonable assumption that anything people will want to do in this callback will need to run on the main event loop.

Reply:
Good point! I'll change it.